rest/Endpoints.ts

```diff
@@ -363,6 +363,67 @@ export default class Endpoints {
     return `/guilds/${guildID}/templates/${code}`;
   }
 
+  // Threads
+  /**
+   * `/channels/{channel.id}/messages/{message.id}/threads`
+   * - POST - Create new thread from existing message
+   */
+  public static START_THREAD_WITH_MESSAGE(channelID: string, messageID: string) {
+    return `/channels/${channelID}/messages/${messageID}/threads`;
+  }
+  /**
+   * `/channels/{channel.id}/threads`
+   * - POST - Creates a new thread that is not connected to an existing message
+   */
+  public static START_THREAD_WITHOUT_MESSAGE(channelID: string) {
+    return `/channels/${channelID}/threads`;
+  }
+  /**
+   * `/channels/{channel.id}/thread-members/@me`
+   * - PUT - Adds the current user to a thread
+   * - DELETE - Removes the current user from a thread
+   */
+  public static THREAD(channelID: string) {
+    return `/channels/${channelID}/thread-members/@me`;
+  }
+  /**
+   * `/channels/{channel.id}/thread-members/{user.id}`
+   * - PUT - Adds another member to a thread
+   * - DELETE - Removes another member from a thread
+   */
+  public static THREAD_MEMBER(channelID: string, userID: string) {
+    return `/channels/${channelID}/thread-members/${userID}`;
+  }
```

Comment:
I'm tempted to say combine these, cc @Khaaz?

Reply:
I'm not sure, what did we do for other similar endpoints?
I'm not against specific /@me endpoint, although I do agree it's legitimate to combine.